src/ansible_navigator/configuration_subsystem/navigator_post_processor.py

```diff
@@ -354,86 +404,101 @@ def execution_environment_volume_mounts(
                     exit_messages.append(ExitMessage(message=exit_msg))
-                    if entry.cli_parameters:
-                        exit_msg = (
-                            "Try again with format "
-                            + f"'{entry.cli_parameters.short}"
-                            + " <source-path>:<destination-path>:<label(Z or z)>'."
-                            + " Note: label is optional."
-                        )
-                        exit_messages.append(ExitMessage(message=exit_msg, prefix=ExitPrefix.HINT))
-                    return messages, exit_messages
-
-            entry.value.current = volume_mounts
+                    exit_messages.append(ExitMessage(message=hint, prefix=ExitPrefix.HINT))
 
-        elif entry.value.current is not C.NOT_SET:
-            parsed_volume_mounts = []
-            volume_mounts = to_list(entry.value.current)
-            for mount_obj in volume_mounts:
-                if not isinstance(mount_obj, dict):
+                try:
+                    volume_mounts.append(
+                        VolumeMount(
+                            fs_source=src,
+                            fs_destination=dest,
+                            options_string=labels,
+                            settings_entry=entry_name,
+                            source=entry_source,
+                        ),
+                    )
+                except VolumeMountError as exc:
                     exit_msg = (
-                        "The following execution-environment.volume-mounts"
-                        f" entry could not be parsed: {mount_obj}"
+                        f"The following {entry_name} entry could not be parsed:"
+                        f" {mount_str} ({entry.value.source.value}). Errors were found: {str(exc)}"
                     )
                     exit_messages.append(ExitMessage(message=exit_msg))
-                    if entry.cli_parameters:
+                    exit_messages.append(ExitMessage(message=hint, prefix=ExitPrefix.HINT))
+
+        elif entry.value.source is C.USER_CFG:
+            hint = (
+                "The value of execution-environment.volume-mounts should be a list"
+                " of dictionaries and valid keys are 'src', 'dest' and 'label'."
+            )
+            if not isinstance(entry.value.current, list):
+                exit_msg = f"{entry_name} entries could not be parsed. ({entry_source.value})"
+                exit_messages.append(ExitMessage(message=exit_msg))
+                exit_msg = hint
+                exit_messages.append(ExitMessage(message=exit_msg, prefix=ExitPrefix.HINT))
+            else:
+                for volume_mount in entry.value.current:
+                    try:
+                        volume_mounts.append(
+                            VolumeMount(
+                                fs_source=volume_mount.get("src"),
+                                fs_destination=volume_mount.get("dest"),
```

Comment:
`.get()` will return `None` if the key doesn't exist, but the type of `VolumeMount`'s `fs_source` and `fs_destination` say these can't be optional. This shouldn't typecheck..?

Reply:
entry.value.current is a List of Any
volume_mount is then an Any
so
volume_mount.get("src") is an Any

which satisfies fs_source being a string

Reply:
If entry value current was something else, I think it would cascade all the way down...